ddtrace/_trace/utils_botocore/aws_payload_tagging.py

```diff
@@ -0,0 +1,242 @@
+        "$..PhoneNumber",
+        # EventBridge
+        "$..AuthParameters.BasicAuthParameters.Password",
+        "$..AuthParameters.OAuthParameters.ClientParameters.ClientSecret",
+        "$..AuthParameters.ApiKeyAuthParameters.ApiKeyValue",
+        # S3
+        "$..SSECustomerKey",
+        "$..CopySourceSSECustomerKey",
+        "$..RestoreRequest.OutputLocation.S3.Encryption.KMSKeyId",
+    ]
+
+    _RESPONSE_REDACTION_PATHS_DEFAULTS = [
+        # // Sns
+        "$..Endpoints.*.Token",
+        "$..PlatformApplication.*.PlatformCredential",
+        "$..PlatformApplication.*.PlatformPrincipal",
+        "$..Subscriptions.*.Endpoint",
+        "$..PhoneNumbers[*].PhoneNumber",
+        "$..phoneNumbers[*]",
+        # // S3
+        "$..Credentials.SecretAccessKey",
+        "$..Credentials.SessionToken",
+    ]
+
+    def __init__(self):
+        self.current_tag_count = 0
+        self.validated = False
+        self.request_redaction_paths = None
+        self.response_redaction_paths = None
+
+    def expand_payload_as_tags(self, span: Span, result: Dict[str, Any], key: str) -> None:
+        """
+        Expands the JSON payload from various AWS services into tags and sets them on the Span.
+        """
+        if not self.validated:
+            self.request_redaction_paths = self._get_redaction_paths_request()
+            self.response_redaction_paths = self._get_redaction_paths_response()
+            self.validated = True
+
+        if not self.request_redaction_paths and not self.response_redaction_paths:
+            return
+
+        if not result:
+            return
+
+        # we will be redacting at least one of request/response
+        redacted_dict = copy.deepcopy(result)
+        self.current_tag_count = 0
+        if self.request_redaction_paths:
+            self._redact_json(redacted_dict, span, self.request_redaction_paths)
+        if self.response_redaction_paths:
+            self._redact_json(redacted_dict, span, self.response_redaction_paths)
+
+        # flatten the payload into span tags
+        for key2, value in redacted_dict.items():
+            escaped_sub_key = key2.replace(".", "\\.")
+            self._tag_object(span, f"{key}.{escaped_sub_key}", value)
+            if self.current_tag_count >= config.botocore.get("payload_tagging_max_tags"):
+                return
+
+    def _should_json_parse(self, obj: Any) -> bool:
+        if isinstance(obj, (str, bytes)):
+            return True
+        return False
+
+    def _validate_json_paths(self, paths: Optional[str]) -> bool:
+        """
+        Checks whether paths is "all" or all valid JSONPaths
+        """
+        if not paths:
+            return False  # not enabled
+
+        if paths == "all":
+            return True  # enabled, use the defaults
+
+        # otherwise validate that we have valid JSONPaths
+        for path in paths.split(","):
+            if path:
+                try:
+                    parse(path)
```

Comment:
We're throwing away the parsing result here. Any chance we can save it to avoid having to re-parse again?

Reply:
Yeah that is a good idea, I'll make a task for it to cache the parsed expressions (when they are all validated) and then use that in `_redact_json`